client/src/app.tsx

```diff
@@ -27,13 +29,20 @@ class App extends Component<{}, State> {
             this.setState({content, contentDate: Date.now(), hasError: false});
             window.scrollTo(0, 0);
         });
+        this.setContext();
     }
 
     componentDidCatch(err: Error, info: any) { // eslint-disable-line class-methods-use-this
         log.error('Error rendering ', {err, info});
         this.setState({hasError: true});
     }
 
+    setContext() {
+        api.context().then((context: Context) => {
```

Comment:
Let's use the `await` syntax here instead of calling `.then`

Reply:
Do you mean to block the app loading before the context loaded from backend?

Reply:
Something like...

```
async setContext() {
    const context = await api.context();
    setContext(context)
}
```

Reply:
Oh, actually... I think this introduces a race condition. What happens if someone calls `getContextItem` before this api request completes? I think that we'll need to make `getContextItem` async and callers will need to await on it.

Reply:
Good call. But as we don't want to give users the switch option, I think this is no longer needed. I will clean it up